meterfs/meterfs.c

```diff
@@ -305,11 +312,255 @@ static int meterfs_eraseFileTable(unsigned int n, meterfs_ctx_t *ctx)
 }
 
 
+static int meterfs_isHeaderValid(unsigned int headerIdx, index_t *id, unsigned int *filecnt, bool *mustMigrate, meterfs_ctx_t *ctx)
+{
+	ssize_t err;
+	union {
+		header_t h;
+		fileheader_t f;
+		headerOld_t hOld;
+		fileheaderOld_t fOld;
+	} u;
+	bool headerOk = false;
+	bool migrate = false;
+
+	err = ctx->read(ctx->devCtx, ctx->offset + (headerIdx * ctx->h1Addr), &u.h, sizeof(u.h));
+	if (err < 0) {
+		meterfs_powerCtrl(0, ctx);
```

Comment:
It would be much more readable, if powerctrl(0) would be call from the place where powerctrl(1) is called, in error paths.

Would be great to mark somehow functions that needs powerctrl.